Cross-deployments guide: fix listener and API guidance and add link to cross-dag learn guide (#3726)
astronomer/docs: astro/best-practices/cross-deployment-dependencies.md

1. In your upstream Deployment, which is the Deployment for which you did **not** create an API Token, use **Variables** in the Astro UI to create an environment variable for your API token, and use `API_TOKEN` for the key.
2. For your downstream Deployment, follow the guidance in [Airflow REST API](https://docs.astronomer.io/astro/airflow-api#step-2-retrieve-the-deployment-url) to obtain the Deployment URL for your downstream Deployment. 
3. In your upstream Deployment, use Variables in the Astro UI to create an environment variable where you can store your downstream Deployment URL, using `DEPLOYMENT_URL` for the key.
4. In the upstream Deployment, implement a listener following the guidance in the [Create Airflow listeners tutorial](https://docs.astronomer.io/learn/airflow-listeners). Use the following code in place of the `listeners_code.py` code provided there.

:::note

> Although this listener is designed to listen only for updates to the specific dataset in this example, the conditional statement `if dataset.uri ...` can be removed or adjusted if you would like to listen for updates to multiple datasets or all datasets in a deployment.

Reviewer: ```suggestion
Although this listener is designed to make an API request only for updates to the specific dataset in this example, the conditional statement `if dataset.uri ...` can be removed or adjusted if you would like to apply the function for updates to multiple datasets or all datasets in a deployment.
```
Since technically the listener does "listen" for everything, and then within it we do the filtering. Unlikely to cause confusion in practice, but just adding that clarity.

@hookimpl
def on_dataset_changed(dataset: Dataset):
    """

>     Execute if a dataset is updated. Remove or adjust the if statement 

Reviewer: ```suggestion
    Execute if a specific dataset is updated. Remove or adjust the if statement 
```

>     to listen for multiple datasets.

Reviewer: ```suggestion
    to apply the function to multiple datasets.
```
